src/util/hooks-helpers.ts

```diff
@@ -43,6 +43,26 @@ interface UseClickListenerParams {
     onOutsideClick?: (event: MouseEvent, targetIsAButton: boolean) => void;
 }
 
+const isButton = (buttonCandidate: any) : boolean => {
```

Comment:
I'm happy with the code, but can you please rename the method into `elementIsInsideButton` or similar to emphasize the fact that parents are checked as well?

Reply:
Sure, I changed it.